File: kotlin/src/main/kotlin/com/tomoarrow/idv/bff/controller/MockController.kt
package com.tomoarrow.idv.bff.controller

import com.tomoarrow.idv.bff.service.MockService
import com.tomoarrow.idv.bff.service.TokenService
import kotlinx.coroutines.runBlocking
import org.springframework.web.bind.annotation.*

@RestController
class MockController(
    private val mockService: MockService,
    private val tokenService: TokenService
) {

    @PostMapping("/v1/idv/cn/mock/start")
    fun idvCnMockStart(@RequestBody body: Map<String, Any>): Any = runBlocking {
        val token = tokenService.requireAccessToken()
        mockService.idvCnMockStart("Bearer $token", body)
    }

    @PostMapping("/v1/idv/cn/mock/token")
    fun idvCnMockToken(@RequestBody body: Map<String, Any>): Any = runBlocking {
        val token = tokenService.requireAccessToken()
        mockService.idvCnMockToken("Bearer $token", body)
    }

    @PostMapping("/v1/idv/cn/mock/kyc/get")
    fun idvCnMockKycGet(@RequestBody body: Map<String, Any>): Any = runBlocking {
        val token = tokenService.requireAccessToken()
        mockService.idvCnMockKycGet("Bearer $token", body)
    }
}

File: typescript/src/mock/mock.controller.ts
import { Body, Controller, Post } from '@nestjs/common';
import { MockService } from './mock.service';
import type { CnMockStartBody, CnMockUserIdBody } from './mock.service';
import { rethrowUpstream } from '../upstream-response';

@Controller()
export class MockController {
  constructor(private readonly mockService: MockService) {}

  // ── CN Mock ──

  @Post('/v1/idv/cn/mock/start')
  async idvMockStartCN(@Body() body: CnMockStartBody): Promise<any> {
    try {
      return await this.mockService.idvMockStartCN(body);
    } catch (e) {
      return rethrowUpstream(e);
    }
  }

  @Post('/v1/idv/cn/mock/token')
  async idvMockTokenCN(@Body() body: CnMockUserIdBody): Promise<any> {
    try {
      return await this.mockService.idvMockTokenCN(body);
    } catch (e) {
      return rethrowUpstream(e);
    }
  }

  @Post('/v1/idv/cn/mock/kyc/get')
  async idvMockKycGetCN(@Body() body: CnMockUserIdBody): Promise<any> {
    try {
      return await this.mockService.idvMockKycGetCN(body);
    } catch (e) {
      return rethrowUpstream(e);
    }
  }
}
